wicket-core/src/main/java/org/apache/wicket/coep/CoepConfiguration.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.coep;
+
+import org.apache.wicket.coop.CoopConfiguration;
+import org.apache.wicket.request.http.WebResponse;
+
+import java.util.Arrays;
+import java.util.HashSet;
+
+/**
+ * Specifies the configuration for Cross-Origin Embedder Policy to be used for
+ * {@link CoepRequestCycleListener}. Users can specify the paths that should be exempt from COEP and
+ * one of 2 modes (<code>REPORTING, ENFORCING</code>) for the policy.
+ *
+ * You can enable COEP headers by adding it to the request cycle listeners in your
+ * {@link org.apache.wicket.protocol.http.WebApplication#init() application's init method}:
+ *
+ * <pre>
+ * &#064;Override
+ * protected void init()
+ * {
+ * 	// ...
+ * 	enableCoep(new CoepConfiguration.Builder().withMode(CoepMode.ENFORCING)
+ * 		.withExemptions("EXEMPTED PATHS").build());
+ * 	// ...
+ * }
+ * </pre>
+ *
+ * @author Santiago Diaz - saldiaz@google.com
+ * @author Ecenaz Jen Ozmen - ecenazo@google.com
+ *
+ * @see CoepRequestCycleListener
+ */
+public class CoepConfiguration
+{
+	public enum CoepMode
+	{
+		ENFORCING("Cross-Origin-Embedder-Policy"), REPORTING(
+			"Cross-Origin-Embedder-Policy-Report-Only");
+
+		final String header;
+
+		CoepMode(String header)
+		{
+			this.header = header;
+		}
+	}
+
+	static final String REQUIRE_CORP = "require-corp";
+
+	private final HashSet<String> exemptions;
```

Comment:
Even though exemptions is an instance of `HashSet`, it is better to make its type be `Set<String>`